internal/provider/mssql_permission_resource.go

```diff
@@ -0,0 +1,171 @@
+package provider
```

Comment:
1. Needs at least an example or a test
2. Im not sure what a permission here is - is this grants?
3. Can we keep it to one new resource per PR?